metadata/src/lib.rs

```diff
@@ -3,844 +3,120 @@
-    pallet: &frame_metadata::PalletMetadata<PortableForm>,
-) -> [u8; 32] {
-    // Begin with some arbitrary hash (we don't really care what it is).
-    let mut bytes = hash(&[19]);
-    let mut visited_ids = HashSet::<u32>::new();
-
-    if let Some(calls) = &pallet.calls {
-        bytes = xor(
-            bytes,
-            get_type_hash(registry, calls.ty.id, &mut visited_ids),
-        );
-    }
-    if let Some(ref event) = pallet.event {
-        bytes = xor(
-            bytes,
-            get_type_hash(registry, event.ty.id, &mut visited_ids),
-        );
-    }
-    for constant in pallet.constants.iter() {
-        bytes = xor(bytes, hash(constant.name.as_bytes()));
-        bytes = xor(
-            bytes,
-            get_type_hash(registry, constant.ty.id, &mut visited_ids),
-        );
-    }
-    if let Some(ref error) = pallet.error {
-        bytes = xor(
-            bytes,
-            get_type_hash(registry, error.ty.id, &mut visited_ids),
-        );
-    }
-    if let Some(ref storage) = pallet.storage {
-        bytes = xor(bytes, hash(storage.prefix.as_bytes()));
-        for entry in storage.entries.iter() {
-            bytes = hash_hashes(
-                bytes,
-                get_storage_entry_hash(registry, entry, &mut visited_ids),
-            );
-        }
-    }
-
-    bytes
-}
-
-/// Obtain the hash representation of a `frame_metadata::RuntimeMetadataV14`.
-pub fn get_metadata_hash(metadata: &RuntimeMetadataV14) -> [u8; 32] {
-    // Collect all pairs of (pallet name, pallet hash).
-    let mut pallets: Vec<(&str, [u8; 32])> = metadata
-        .pallets
-        .iter()
-        .map(|pallet| {
-            let hash = get_pallet_hash(&metadata.types, pallet);
-            (&*pallet.name, hash)
-        })
-        .collect();
-
-    // Sort by pallet name to create a deterministic representation of the underlying metadata.
-    pallets.sort_by_key(|&(name, _hash)| name);
-
-    // Note: pallet name is excluded from hashing.
-    // Each pallet has a hash of 32 bytes, and the vector is extended with
-    // extrinsic hash and metadata ty hash (2 * 32).
-    let mut bytes = Vec::with_capacity(pallets.len() * 32 + 64);
-    for (_, hash) in pallets.iter() {
-        bytes.extend(hash)
-    }
-
-    bytes.extend(get_extrinsic_hash(&metadata.types, &metadata.extrinsic));
 
-    let mut visited_ids = HashSet::<u32>::new();
-    bytes.extend(get_type_hash(
-        &metadata.types,
-        metadata.ty.id,
-        &mut visited_ids,
-    ));
-
-    hash(&bytes)
-}
+pub use retain::retain_metadata_pallets;
+pub use validation::{
+    get_call_hash, get_constant_hash, get_metadata_hash, get_metadata_per_pallet_hash,
+    get_pallet_hash, get_storage_hash, NotFound,
+};
 
-/// Obtain the hash representation of a `frame_metadata::RuntimeMetadataV14`
-/// hashing only the provided pallets.
+/// Latest runtime metadata version supported by subxt.
+pub type LatestRuntimeMetadata = RuntimeMetadataV15;
```

Comment:
I wouldn't bother with this alias (and it's not used in a bunch of places where it could be anyway)